tests/backends/aiida_django/migrations/test_migrations_0046_inline_calc_attributes.py

```diff
@@ -0,0 +1,98 @@
+###########################################################################
+# pylint: disable=import-error,no-name-in-module,invalid-name
+"""Test the migration that updates the attributes of InlineFunctions that have been transformed into CalcFunctions."""
+from aiida.backends.general.migrations import utils
+from .test_migrations_common import TestMigrations
+
+
+class TestCalcFunctionAttributesMigration(TestMigrations):
+    """Test the migration for the attributes of InlineFunctions that have been transformed into CalcFunctions."""
+
+    migrate_from = '0045_dbgroup_extras'
+    migrate_to = '0046_inline_calc_attributes'
+
+    @staticmethod
+    def get_source_code(full_file=False):
+        """Returns a string containing the source code for the calcfunction"""
+
+        # yapf: disable
+        text_i = ('from aiida import orm\n'
+                  'from aiida.engine import calcfunction\n'
+        )
+        text_t = ('@calcfunction\n'
+                  'def sum_inline( int1, int2 ):\n'
+                  '    return int1 + int2\n'
+        )
+        text_f = ('num1 = orm.Int(2)\n'
+                  'num2 = orm.Int(3)\n'
+                  'num3 = sum_inline( int1, int2 )\n'
+                  'print(num3)\n'
+        )
+        # yapf: enable
+
+        if full_file:
+            text_t = text_i + '\n' + text_t + '\n' + text_f
+
+        return text_t
+
+    def setUpBeforeMigration(self):
+        calcnode = self.DbNode(
+            node_type='process.calculation.calcfunction.CalcFunctionNode.',
+            user_id=self.default_user.id,
+            attributes={
+                'sealed': True,
+                'namespace': '__main__',
+                'source_code': self.get_source_code(full_file=False),
+                'source_file': self.get_source_code(full_file=True),
+                'function_name': 'sum_inline',
+                'first_line_source_code': 4,
+            },
+        )
+        calcnode.save()
+        self.calcnode_id = calcnode.id
+        #print(self.load_node(self.calcnode_id).attributes)
+        #calcnode.attributes['function_namespace'] = calcnode.attributes['namespace']
+        #calcnode.save()
+        #print(self.load_node(self.calcnode_id).attributes)
+
+    def test_data_migrated(self):
+        """Test that the model now has an extras column with empty dictionary as default."""
+        calcnode = self.load_node(self.calcnode_id)
+
+        # Pre existing and renamed attributes
+        self.assertEqual(calcnode.attributes['sealed'], True)
+        self.assertEqual(calcnode.attributes['function_name'], 'sum_inline')
+        self.assertEqual(calcnode.attributes['function_namespace'], '__main__')
+        self.assertEqual(calcnode.attributes['function_starting_line_number'], 4)
+
+        # Newly determined attributes
+        self.assertEqual(calcnode.attributes['exit_status'], 0)
+        self.assertEqual(calcnode.attributes['process_state'], 'finished')  #USE ENUM?
```

Comment:
If you do decide to use the enum, then you should probably use it also consistently in the migration itself. There you also just use the string. I think it is actually fine to use the string here.
```suggestion
        self.assertEqual(calcnode.attributes['process_state'], 'finished')
```

Reply:
Ah, yes, indeed. So this was a question: I was looking if the `process_state` was set as an enum but couldn't find / didn't know how to grep for it. Do you know where it is? I should probably use that, right? (as you say, both in test and in migration)

Reply:
it comes from the `plumpy.ProcessState` enum, which you can import from `aiida.engine.processes.process` as it is aliased there

Reply:
Ok, thanks! I've actually changed my mind...since the migrations apply to the database, and the database at that point had the `FINISHED` state be `"finished"`, I think the correct way to do this is to target that specific version (i.e. if we later change our `FINISHED` state to `"totally_finished"`, it may break this migration because at the DB where it is applying it the values stored are `"finished"` and not `"totally_finished"`).